thriftsimple/thriftsimple.go

```diff
@@ -0,0 +1,160 @@
+package thriftsimple
+
+import (
+	"context"
+	"errors"
+	"io/ioutil"
+	"time"
+
+	"github.com/apache/thrift/lib/go/thrift"
+	"github.com/getsentry/raven-go"
+	"github.com/reddit/baseplate.go/edgecontext"
+	"github.com/reddit/baseplate.go/log"
+	"github.com/reddit/baseplate.go/metricsbp"
+	"github.com/reddit/baseplate.go/secrets"
+	"github.com/reddit/baseplate.go/thriftbp"
+	"github.com/reddit/baseplate.go/tracing"
+	"gopkg.in/yaml.v2"
+)
+
+type BaseplateServerConfig struct {
+	Addr string
+
+	Timeout time.Duration
+
+	Debug bool
+
+	Metrics struct {
+		Namespace string
+		Endpoint  string
+	}
+
+	Secrets struct {
+		Path string
+	}
+
+	Sentry struct {
+		DSN         string
+		Environment string
+		SampleRate  float64
+	}
+
+	Tracing struct {
+		Namespace     string
+		Endpoint      string
+		RecordTimeout time.Duration `yaml:"recordTimeout"`
+		SampleRate    float64
+	}
+}
+
+func (c BaseplateServerConfig) AsConfig() BaseplateServerConfig {
```

Comment:
What is this for?

Reply:
If you embed BaseplateServerConfig into a larger struct, then this let's you cast back into a BaseplateServerConfig